Gem/Nodes/Scenes/ImportNodeScene.swift

```diff
@@ -0,0 +1,162 @@
+                    .bold()
+            }
+            ToolbarItem(placement: .navigationBarTrailing) {
+                Button(action: onSelectScan) {
+                    Image(systemName: SystemImage.qrCode)
+                }
+                .bold()
+            }
+        }
+    }
+}
+
+// MARK: - UI Components
+
+extension ImportNodeScene {
+    private var inputSection: some View {
+        Section {
+            HStack {
+                FloatTextField(Localized.Settings.Networks.ImportNode.rpcAddress, text: $model.urlString)
+                    .textFieldStyle(.plain)
+                    .focused($focusedField, equals: .address)
+                    .textInputAutocapitalization(.never)
+                    .autocorrectionDisabled()
+                    .submitLabel(.search)
+                    .onSubmit(onSubmitUrl)
+                Spacer()
+                HStack(spacing: Spacing.medium) {
+                    ListButton(image: Image(systemName: SystemImage.paste), action: onSelectPaste)
+                    ListButton(image: Image(systemName: SystemImage.qrCode), action: onSelectScan)
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var nodeInfoView: some View {
+        switch model.state {
+        case .noData, .loading:
+            EmptyView()
+        case .loaded:
+            Section {
+                ListItemView(
+                    title: Localized.Settings.Networks.ImportNode.chainId,
+                    titleStyle: .body,
+                    subtitle: model.chainID,
+                    subtitleStyle: .calloutSecondary
+                )
+                ListItemView(
+                    title: Localized.Settings.Networks.ImportNode.inSync,
+                    titleStyle: .body,
+                    subtitle: model.isInSync ? Localized.Common.yes : Localized.Common.no,
+                    subtitleStyle: .calloutSecondary
+                )
+                if let blockNumber = model.blockNumber {
+                    ListItemView(
+                        title: Localized.Settings.Networks.ImportNode.latestBlock,
+                        titleStyle: .body,
+                        subtitle: blockNumber,
+                        subtitleStyle: .calloutSecondary
+                    )
+                }
+            }
+        case .error(let error):
+            Section {
+                StateErrorView(error: error, message: error.localizedDescription)
+            }
+        }
+    }
+}
+
+// MARK: - Actions
+
+extension ImportNodeScene {
+    private func onSelectDone() {
+        dismiss()
+    }
+
+    private func onSubmitUrl() {
+        getNetwrokInfoAsync()
+    }
+
+    private func onSelectPaste() {
+        guard let content = UIPasteboard.general.string else {
+            return
+        }
+        model.urlString = content.trim()
+        getNetwrokInfoAsync()
+    }
+
+    private func onSelectImport() {
+        model.importFoundNode()
+        onDismiss()
```

Comment:
dismiss should be called once? `onDismiss` and handeled from the outside of this screen

Reply:
make sence, will do